### ISSUE TITLE
Tracking: Player performance tracking for Windows

### ISSUE THREAD

**Issue by com1clyf3** [CompletelyFairGames/dwarfcorp#711]
Thanks for reporting your performance! Follow the steps below.

**1. What's your specific OS version and hardware specs:** 

If you need help in how find this info, [follow this guide](https://www.wikihow.com/Check-Computer-Specifications) and report all the info it gives you. There will likely be a file produced with all the info for your specs you can drag and drop into your report.

**2. FPS & Describe the issue:** 

_Briefly describe how your game is behaving._ Is the entire game laggy, or do you experience lag only after you perform a specific action? Is there a specific set of steps you have found that create the issue?

_Press ~_ and your performance display should pop up. Your FPS will be found under "Frame Time". Go and perform whatever action is creating your performance issue, and keep an eye on this number. Afterwards, describe the range of your FPS. 

ie "At start game it's 30FPS, after an hour it's more like 20 FPS, and when battling it drops to 0 FPS for about five seconds, and then goes back to 20 FPS."

**3. Build number:**

Each DwarfCorp build is tracked by its githash and/or release number, and plenty of performance issues are unique to certain builds. The easiest way to find this info is by opening the game, and looking at the bottom right of the start screen. There you should see the release number, like this:

![bloop](https://user-images.githubusercontent.com/6562812/40013063-8816bbf8-577a-11e8-9454-d87c125b707b.jpg)

And that's pretty much it. We might have some follow up questions for some of you, but otherwise you can drop off info at any time, and it'll be a giant help for the game! Thanks for helping making DwarfCorp a more stable thingy.

**Comment by croftsr** [comment]
I'm not sure if I'm placing the information in the right place, but please let me know if it isn't in the right place for you guys to notice.

OS and Hardware Specs:
OS - Windows 10 Home edition v. 1803 (x64 bit), x64-based processor
Pen&touch - touch support w/ 10 touch points
Processor - Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz 2.40GHz
RAM - 8.00 GB 

FPS/Problem: 
When paused, Frame rate is at an average of 35 fps. But when time resumes at any speed, fps drops to 1-2 fps regardless of graphics settings from Lowest to Highest.

Build number: 18.08.05-XNA(6634ddcc)

**Comment by mreguero** [comment]
I'm having issues not even 1 hour into gameplay.

OS: Windows 10 home version 1803 64 bit

Processor: Intel Core i7 6700k @ 4GHz
RAM: 16 GB

Game version is 19.07.05_XNA(b0b48e20)

The issue can be clearly seen in the image posted, aparently too many birds have spawned and competely freeze up the game, dwarfs don't move or do anything. The only thing trying to move is the birds.

FPS is pretty much 0 i have to pause the game to be able to click or attempt to make any actions.

Is there any more info i can add to help find the issue?

![20190707171711_1](https://user-images.githubusercontent.com/3283855/60770633-74d76300-a0dd-11e9-9150-7d550482e950.jpg)

**Comment by Blecki** [comment]
Holy shit that's a lot of birds.

**Comment by mreguero** [comment]
yeah, btw if you wait longer they keep coming. I think we found out how many birds is too many birds:P
![20190707183718_1](https://user-images.githubusercontent.com/3283855/60771304-588bf400-a0e6-11e9-9089-33d736133fc0.jpg)

PS: this is another new game.

I don't know if it helps but i use x3 time a lot

**Comment by Blecki** [comment]
Looking into it now.

**Comment by Blecki** [comment]
Okay - they have a species limit of 50, and I've confirmed that still functions. Are you saving and reloading during this?

**Comment by mreguero** [comment]
No loading and on last playthrough only autosave.

**Comment by Blecki** [comment]
So get this... it checks the population limit before laying an egg, but not when it hatches. But the unhatched egg doesn't count against the population limit.

This would allow birds to get to roughly twice the population limit... not quite as bad as you have it, so it's not the whole story.

**Comment by mreguero** [comment]
Searching for a solution before adding the comment I found someone talking about migration, and the birds appear to be comming from the far right outer edge, could this be related?

Also i don't seem to remember seeing any eggs and the birds just keep apearing without any eggs in sight. I'll try to start a new save and wait to see if i see any eggs.

**Comment by mreguero** [comment]
here is the steam issue where it was mentioned seems like the author of the issue had a similar problem.

https://steamcommunity.com/app/252390/discussions/1/1692659135918853171/

**Comment by Blecki** [comment]
I was able to duplicate the problem and fix a bug somewhere else, so it's at least not only migration. I was able to get from 17 birds to 2053 overnight. Problem was really that every egg that got laid hatched into 20+ birds. Fix is available on steam as long as you're running stable. Let me know if you see it again!

**Comment by mreguero** [comment]
sweet i'll download and test right now, thanks!

**Comment by merticois** [comment]
![image](https://user-images.githubusercontent.com/39864292/62006812-cd65b300-b113-11e9-951a-276f05c9088c.png)

I have (I think) the latest update on steam. My game was fine (although performing a bit slowly) when I saved and quit yesterday. When I loaded today there was a bird/egg explosion in my world.

**Comment by Blecki** [comment]
Hit ~ and look for a list of species counts. How many birds does it say you
have?

**Comment by merticois** [comment]
I updated my original post with that shot, not sure how to interpret it all but it looks like about 149 birds? 5468 entity updates though so I am not sure why that number is so big. My performance kept dropping to barely intractable levels. This didn't happen in the desert biome but in the forest it's all over the place. Eggs everywhere too.

**Comment by Blecki** [comment]
Species limit is 50, so it looks like that's working as intended (the game has no concept of chickens being birds - as fair as it's concerned, they are completely different).

I'm thinking I'll have to make the limit configurable, as 50 might still be too many for some systems. In the meantime, you could try lowering your entity update distance.

**Comment by merticois** [comment]
I will do so but my system specs are:
GPU: EVGA GTX 1080 FTW2
CPU: Intel i7 4770k
Memory: 16GB DDR3 1600

I think my install is not on a SSD right now but I don't think I am experiencing cache misses because the game isn't utilizing even close to all my ram.

**Comment by Blecki** [comment]
Most likely you're CPU bound and have maxed out a core, with nothing left
the game can currently offload.

**Comment by Beastie33** [comment]
Hi, periodically I come back to this game, originally purchased in 2014.
I've never had a game last more than a session, it always seems to crash or become unplayable, this has been consistent on various PC's and versions of windows.

Finally I figured I should make a report.

**Current system details:**
AMD Ryzen 7 3700X 8-Core Processor 3.60 GHz
32.0 GB RAM
Nvidia GeForce RTX 2080
Windows 10 Pro 64bit 1909 18363.900

**DwarfCorp 20.06.28_XNA(2e12afg4b)**

**FPS & Describe the issue:**
Initially game seems to run fine at a consistent 30fps no matter the graphics settings selected.
The performance will gradually degrade over the course of play and within about an hour of starting a new game it drops severely eventually becoming unplayable at less than 1fps.

Setting Natives to very low seems to make things last a little longer.
I also very frequently get dwarves rapidly switching between tasks and getting stuck in place, seems to be repeatable when there are jobs to place a block and harvest a crop.

I've attached a video, not sure it that helps,

[Dwarfcorp 2020-06-26 13-04-17-1.zip](https://github.com/Blecki/dwarfcorp/files/4850779/Dwarfcorp.2020-06-26.13-04-17-1.zip)

**Comment by Blecki** [comment]
Can you do one thing to confirm something for me?

Sell all your resources on the market. Everything. If the performance returns, I'll know what the issue is.

Also - you have draw distance maxed out don't you? At default distance you can see a total of 66 chunks, you got like 20k.

**Comment by Beastie33** [comment]
Draw distance is fairly low, not sure what it was in the video above but it's currently 42 and that doesn't seem to help much, unless I move far away from the colony, then things pick up a bit.

I've started a new game to see if selling all items improves things when the slowdown starts.

**Comment by Beastie33** [comment]
I played for an hour or so and I wasn't able to recreate a slowdown to test selling all my items.
Apart from the Dwarfs sometimes getting stuck and some temporary slowdowns I had a pretty much trouble free game, I think that's a first..

**Comment by Blecki** [comment]
Any chance you grabbed the update I put up this morning first?

For the other issue; hover over the task in the task list to see what's causing them to fail.

**Comment by Beastie33** [comment]
I think so, yes. Just noticed the update from two hours ago, that would have been active in the game I just played.
